### ISSUE TITLE
Deprecation warnings when calling help(matplotlib)

### ISSUE THREAD

**Issue by astrofrog** [matplotlib/matplotlib#12650]
With the latest developer version (f28897705bb6a98066ea8b3b5716cd25b6ebd078), I'm getting lots of warnings when typing ``help(matplotlib)``:

```
>>> import matplotlib
>>> help(matplotlib)
/Users/tom/miniconda3/envs/alldev/lib/python3.7/inspect.py:422: MatplotlibDeprecationWarning: The msg_backend_obsolete attribute was deprecated in Matplotlib 3.0 and will be removed in 3.2.
  get_obj = getattr(cls, name)
/Users/tom/miniconda3/envs/alldev/lib/python3.7/inspect.py:434: MatplotlibDeprecationWarning: The msg_backend_obsolete attribute was deprecated in Matplotlib 3.0 and will be removed in 3.2.
  srch_obj = getattr(srch_cls, name, None)
/Users/tom/miniconda3/envs/alldev/lib/python3.7/inspect.py:422: MatplotlibDeprecationWarning: The msg_depr attribute was deprecated in Matplotlib 3.0 and will be removed in 3.2.
  get_obj = getattr(cls, name)
/Users/tom/miniconda3/envs/alldev/lib/python3.7/inspect.py:434: MatplotlibDeprecationWarning: The msg_depr attribute was deprecated in Matplotlib 3.0 and will be removed in 3.2.
  srch_obj = getattr(srch_cls, name, None)
/Users/tom/miniconda3/envs/alldev/lib/python3.7/inspect.py:422: MatplotlibDeprecationWarning: The msg_depr_ignore attribute was deprecated in Matplotlib 3.0 and will be removed in 3.2.
  get_obj = getattr(cls, name)
/Users/tom/miniconda3/envs/alldev/lib/python3.7/inspect.py:434: MatplotlibDeprecationWarning: The msg_depr_ignore attribute was deprecated in Matplotlib 3.0 and will be removed in 3.2.
  srch_obj = getattr(srch_cls, name, None)
/Users/tom/miniconda3/envs/alldev/lib/python3.7/inspect.py:422: MatplotlibDeprecationWarning: The msg_depr_set attribute was deprecated in Matplotlib 3.0 and will be removed in 3.2.
  get_obj = getattr(cls, name)
/Users/tom/miniconda3/envs/alldev/lib/python3.7/inspect.py:434: MatplotlibDeprecationWarning: The msg_depr_set attribute was deprecated in Matplotlib 3.0 and will be removed in 3.2.
  srch_obj = getattr(srch_cls, name, None)
/Users/tom/miniconda3/envs/alldev/lib/python3.7/inspect.py:422: MatplotlibDeprecationWarning: The msg_obsolete attribute was deprecated in Matplotlib 3.0 and will be removed in 3.2.
  get_obj = getattr(cls, name)
/Users/tom/miniconda3/envs/alldev/lib/python3.7/inspect.py:434: MatplotlibDeprecationWarning: The msg_obsolete attribute was deprecated in Matplotlib 3.0 and will be removed in 3.2.
  srch_obj = getattr(srch_cls, name, None)
/Users/tom/miniconda3/envs/alldev/lib/python3.7/inspect.py:575: MatplotlibDeprecationWarning: The msg_backend_obsolete attribute was deprecated in Matplotlib 3.0 and will be removed in 3.2.
  if cls is None or getattr(cls, name) is not obj:
/Users/tom/miniconda3/envs/alldev/lib/python3.7/inspect.py:587: MatplotlibDeprecationWarning: The msg_backend_obsolete attribute was deprecated in Matplotlib 3.0 and will be removed in 3.2.
  doc = getattr(base, name).__doc__
/Users/tom/miniconda3/envs/alldev/lib/python3.7/inspect.py:575: MatplotlibDeprecationWarning: The msg_depr attribute was deprecated in Matplotlib 3.0 and will be removed in 3.2.
  if cls is None or getattr(cls, name) is not obj:
/Users/tom/miniconda3/envs/alldev/lib/python3.7/inspect.py:587: MatplotlibDeprecationWarning: The msg_depr attribute was deprecated in Matplotlib 3.0 and will be removed in 3.2.
  doc = getattr(base, name).__doc__
/Users/tom/miniconda3/envs/alldev/lib/python3.7/inspect.py:575: MatplotlibDeprecationWarning: The msg_depr_ignore attribute was deprecated in Matplotlib 3.0 and will be removed in 3.2.
  if cls is None or getattr(cls, name) is not obj:
/Users/tom/miniconda3/envs/alldev/lib/python3.7/inspect.py:587: MatplotlibDeprecationWarning: The msg_depr_ignore attribute was deprecated in Matplotlib 3.0 and will be removed in 3.2.
  doc = getattr(base, name).__doc__
/Users/tom/miniconda3/envs/alldev/lib/python3.7/inspect.py:575: MatplotlibDeprecationWarning: The msg_depr_set attribute was deprecated in Matplotlib 3.0 and will be removed in 3.2.
  if cls is None or getattr(cls, name) is not obj:
/Users/tom/miniconda3/envs/alldev/lib/python3.7/inspect.py:587: MatplotlibDeprecationWarning: The msg_depr_set attribute was deprecated in Matplotlib 3.0 and will be removed in 3.2.
  doc = getattr(base, name).__doc__
/Users/tom/miniconda3/envs/alldev/lib/python3.7/inspect.py:575: MatplotlibDeprecationWarning: The msg_obsolete attribute was deprecated in Matplotlib 3.0 and will be removed in 3.2.
  if cls is None or getattr(cls, name) is not obj:
/Users/tom/miniconda3/envs/alldev/lib/python3.7/inspect.py:587: MatplotlibDeprecationWarning: The msg_obsolete attribute was deprecated in Matplotlib 3.0 and will be removed in 3.2.
  doc = getattr(base, name).__doc__
/Users/tom/miniconda3/envs/alldev/lib/python3.7/pydoc.py:1054: MatplotlibDeprecationWarning: 
examples.directory is deprecated; in the future, examples will be found relative to the 'datapath' directory.
  return cram(stripid(repr(x)), self.maxother)
/Users/tom/miniconda3/envs/alldev/lib/python3.7/pydoc.py:1054: MatplotlibDeprecationWarning: 
examples.directory is deprecated; in the future, examples will be found relative to the 'datapath' directory.
  return cram(stripid(repr(x)), self.maxother)
```

Is this expected?

**Comment by anntzer** [comment]
Likely due to the same cause as https://github.com/matplotlib/matplotlib/pull/12604 (hitting `__getitem__` with Py3.7.1).

**Comment by dstansby** [comment]
Fixed by #12653 and #12658 I think.
